examples/tutorials/introduction/2_initialization.py

```diff
@@ -0,0 +1,175 @@
+have a lot of boilerplate.  In our orignal example, the code copies all of
+the ``__init__`` arguments to corresponding attributes before doing any
+real work::
+
+    def __init__(self, filename, sample_id, date_aquired, operator,
+                 scan_size=(1e-5, 1e-5)):
+        # initialize the primary attributes
+        self.filename = filename
+        self.sample_id = sample_id
+        self.operator = operator
+        self.scan_size = scan_size
+
+Traits lets you avoid this boilerplate by defining a default ``__init__``
+method that accepts keyword arguments that correspond to the declared
+traits.  The Traits version of the ``Image`` class could potentially skip
+the ``__init__`` method entirely::
+
+    class Image(HasTraits):
+        filename = File(exists=True)
+        sample_id = Str()
+        date_acquired = Date()
+        operator = Str()
+        scan_size = Tuple(Float, Float)
+
+
+    # this works!
+    image = Image(
+        filename=filename,
+        operator="Hannes",
+        sample_id="0001",
+        date_acquired=datetime.datetime.today(),
+        scan_size=(1e-5, 1e-5),
+    )
+
+Default Values
+--------------
+
+There are a couple of complications in the example that we need to take into
+account.  The first is what happens if a user forgets to provide an initial
+value::
+
+    >>> image = Image(
+    ...     filename=filename,
+    ...     sample_id="0001",
+    ...     date_acquired=datetime.datetime.today(),
+    ...     scan_size=(1e-5, 1e-5),
+    ... )
+    >>> image.operator
+    ""
+
+As this example shows, the operator gets given a default value of the empty
+string ``""``.  In fact every trait type comes with an default value.  For
+numeric trait types, like ``Int`` and ``Float``, the default is 0.  For
+``Str`` trait types it is the empty string, for ``Bool`` traits it is
+``False``, and so on.
+
+However, that might not be what you want as your default value.  For example,
+you might want to instead flag that the operator has not been provided with
+the string ``"N/A"`` for "not available".  Most trait types allow you to
+specify a default value as part of the declaration.  So we could say::
+
+    operator = Str("N/A")
+
+and now if we omit ``operator`` from the arguments, we get::
+
+    >>> image.operator
+    "N/A"
+
+
+Dynamic Defaults
+----------------
+
+The second complication comes from more complex initial values.  For example,
+we could declare some arbitrary fixed date as the default value for
+``date_acquired``::
+
+    date_acquired = Date(datetime.date(2020, 1, 1))
+
+But it would be better if we could set it to a dynamic value.  For example,
+a reasonable default would be today's date.  You can provide this sort of
+dynamically declared default by using a specially-named method which has
+the pattern ``_<trait-name>_default`` and which returns the default value.
+So we could write::
+
+    def _date_acquired_default(self):
+        return datetime.datetime.today()
+
```

Comment:
This maybe a good place to recommend keeping the dynamic default method independent of other mutable traits. 
The rationale is the possible/maybe-confusing interactions with change notifications. This reasoning involves concepts yet to be introduced though, maybe we can say something like this?:

> It is recommended that dynamic defaults are independent of other traits. Complex dependencies among traits are best handled using traits observations and property, see the next few topics in this tutorial.